README.md

```diff
@@ -1,8 +1,362 @@
+|---------------|-------------------------------|
+| 2.x           | 11.1 - 11.8, 12.x (only latest patch releases) |
+| 3.x (Future)  | 12.x, 13.x  (only latest patch releases)       |
+
+[Well-behaved code](#compatibility-guidelines) using the latest CCCL should compile and run successfully with any supported CTK version. 
+Exceptions may occur for new features that depend on new CTK features, so those features would not work on older versions of the CTK. 
+For example, C++20 support was not added to `nvcc` until CUDA 12.0, so CCCL features that depend on C++20 would not work with CTK 11.x. 
+
+We want to enable users to bring a newer version of CCCL to an older CTK, but not vice versa.
+This means you cannot bring an older version of CCCL to a newer CTK.
+In other words, **CCCL is never forwards compatible with the CUDA Toolkit.** 
+
+The table below summarizes compatibility of the CTK and CCCL:
+
+| CTK Version | Included CCCL Version |   Trying to use...  | Supported? |                            Notes                           |
+|:-----------:|:---------------------:|:-------------------:|:----------:|:----------------------------------------------------------:|
+|   CTK `X.Y`   |    CCCL `MAJOR.MINOR`   | CCCL `MAJOR.MINOR+n`  |     Yes    |               Some new features may not work               |
+|   CTK `X.Y`   |    CCCL `MAJOR.MINOR`   | CCCL `MAJOR+1.MINOR`  |     Yes    | Possible breaking changes.  Some new features may not work |
+|   CTK `X.Y`   |    CCCL `MAJOR.MINOR`   | CCCL `MAJOR+2.MINOR`  |     No     |           CCCL supports only 2 CTK major verions           |
+|   CTK `X.Y`   |    CCCL `MAJOR.MINOR`   | CCCL `MAJOR.MINOR-n`  |     No     |               CCCL is not forwards compatible              |
+|   CTK `X.Y`   |    CCCL `MAJOR.MINOR`   | CCCL `MAJOR-n.MINOR`  |     No     |               CCCL is not forwards compatible              |
+
+
+For more information on CCCL versioning, API/ABI compatibility, and breaking changes see the [Versioning](#versioning) section below.
+
+
+### Operating Systems
+
+Unless otherwise specified, CCCL supports all the same operating systems as the CUDA toolkit, which are documented here:
+ - [Linux](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#system-requirements)
+ - [Windows](https://docs.nvidia.com/cuda/cuda-installation-guide-microsoft-windows/index.html#system-requirements)
+
+### Host Compilers
+
+Unless otherwise specified, CCCL supports all the same host compilers as the CUDA Toolkit, which are documented here:
+- [Linux](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#host-compiler-support-policy)
+- [Windows](https://docs.nvidia.com/cuda/cuda-installation-guide-microsoft-windows/index.html#system-requirements)
+
+### C++ Dialects
+- C++11 (Deprecated in Thrust/CUB, to be removed in next major version)
+- C++14 (Deprecated in Thrust/CUB, to be removed in next major version)
+- C++17 
```

Comment:
Trailing whitespace. Please trim the trailing whitespace in this file, there are other places where this is happening.
```suggestion
- C++17
```